crates/re_renderer/src/renderer/point_cloud.rs

```diff
@@ -725,3 +725,47 @@ impl Renderer for PointCloudRenderer {
         Ok(())
     }
 }
+
+/// Determines the size for a "data texture", i.e. a texture that we use to store data
+/// that we'd otherwise store inside a buffer, but can't due to WebGL compatibility.
+///
+/// `max_texture_size` must be a power of two.
+///
+/// For convenience, the returned texture size has a width that is also a multiple of `wgpu::COPY_BYTES_PER_ROW_ALIGNMENT`
+/// for the given element size.
+/// This makes it a lot easier to copy data from a continuous buffer to the texture.
+/// If we wouldn't do that, we'd need to do a copy per row in some cases.
+fn data_texture_size(
+    num_elements: u32,
+    element_size_in_bytes: u32,
+    max_texture_size: u32,
+) -> wgpu::Extent3d {
+    assert!(max_texture_size.is_power_of_two());
+
+    // Our data textures are usually accessed in a linear fashion, so ideally we'd be using a 1D texture.
+    // However, 1D textures are very limited in size on many platforms, we we have to use 2D textures instead.
+    // 2D textures perform a lot better when their dimensions are powers of two, so we'll strictly stick to that even
+    // when it seems to cause memory overhead.
+
+    // We fill row by row. With the power-of-two requirement, this is the optimal strategy:
+    // if there were a texture with less padding that uses half the width,
+    // then we'd need to increase the height. We can't increase without doubling it, thus creating a texture
+    // with the exact same mount of padding as before ▮.
+
+    let width = if num_elements < max_texture_size {
+        num_elements
+            .next_power_of_two()
+            // Keeping at a multiple of the alignment requirement makes copy operations a lot simpler.
+            .next_multiple_of(wgpu::COPY_BYTES_PER_ROW_ALIGNMENT / element_size_in_bytes)
```

Comment:
This makes no sense to me. Why would the size of the texture _shrink_ when the element size goes up?